### PR TITLE
feat: add websocket watcher

### PR DESCRIPTION
**Pull request by zepatrik**
The websocket watcher is just any of the other watchers but piped through a websocket.

### CODE REVIEW

File: watcherx/websocket_client.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+func closeOnDone(ctx context.Context, conn *websocket.Conn) {
+	<-ctx.Done()
+	conn.Close()
+}
+
+func forwardWebsocketEvents(conn *websocket.Conn, c EventChannel, u *url.URL) {
+	serverURL := source(u.String())
+	defer conn.Close()
+	for {
```
Comment (aeneasr) [inline watcherx/websocket_client.go:29]: This queue never finishes. Maybe listen on `context.Done()`?

Comment (zepatrik) [reply watcherx/websocket_client.go:29]: That's why there is https://github.com/ory/x/pull/190/files#diff-5db460a2bf191856555ebe1ceb20c2b3R16

See https://github.com/gorilla/websocket/issues/71 as a reference.

Comment (zepatrik) [reply watcherx/websocket_client.go:29]: `conn.ReadMessage` is blocking

Comment (aeneasr) [reply watcherx/websocket_client.go:29]: So `WatchWebsocket` will just accumulate go routines that can not be killed? Not sure if I like that :D

**Comment by aeneasr** [comment]
fixed on master you can rebase

**Comment by zepatrik** [comment]
Done

**Review by aeneasr — APPROVED** [review]

**Pull request merged**